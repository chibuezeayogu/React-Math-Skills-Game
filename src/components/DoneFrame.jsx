import React from 'react';
import PropTypes from 'prop-types';

const DoneFrame = ({ doneStatus, resetGame }) => (

  <div className="text-center">
    <h2>{doneStatus}</h2>
      <button className="btn btn-outline-success"
        onClick={resetGame}
      >
      Play Again
    </button>
  </div>
);

DoneFrame.propTypes = {
  doneStatus: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.string
  ]),
  resetGame: PropTypes.func,
};

export default DoneFrame;
