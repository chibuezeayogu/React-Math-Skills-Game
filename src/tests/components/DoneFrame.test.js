/* 
global jest 
expect 
*/
import React from 'react';
import { shallow } from 'enzyme';

import DoneFrame from '../../components/DoneFrame';

describe('DoneFrame Component', () => {
  const setup = () => {
    const props = {
      doneStatus: "Nice Well Done",
      resetGame: jest.fn(),
    };

    return shallow(<DoneFrame {...props}/>);
  };
  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
