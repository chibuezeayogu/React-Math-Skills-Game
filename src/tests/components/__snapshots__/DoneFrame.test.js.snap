// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoneFrame Component should render correctly 1`] = `
<div
  className="text-center"
>
  <h2>
    Nice Well Done
  </h2>
  <button
    className="btn btn-outline-success"
    onClick={[MockFunction]}
  >
    Play Again
  </button>
</div>
`;
